tests/e2e/koperator_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/yaml"
+)
+
+// requireApplyingKoperatorCRDs deploys the koperator CRDs and checks their
+// existence afterwards.
+func requireApplyingKoperatorCRDs(kubectlOptions *k8s.KubectlOptions, koperatorVersion Version) {
+	It("Applying koperator CRDs", func() {
+		By(fmt.Sprintf("Retrieving Koperator CRDs (to work around too long CRD) with version %s ", koperatorVersion))
+		// Note: had to go lower because none of apply, create, replace works
+		// properly with the KafkaCluster CRD due to the size (apply cannot
+		// handle the size, create is not idempotent, replace is only idempotent
+		// if the resources are existing already). Tried dropping the
+		// descriptions in the CRDs, but still too large
+		var rawKoperatorCRDsManifest []byte
+		var err error
+
+		switch koperatorVersion {
+		case LocalVersion:
+			rawKoperatorCRDsManifest = []byte(helm.RenderTemplate(
+				GinkgoT(),
+				&helm.Options{
+					SetValues: map[string]string{
+						"crd.enabled": "true",
+					},
+				},
+				"../../charts/kafka-operator",
+				"dummy",
+				[]string{"templates/crds.yaml"},
+			))
+		default:
+			httpClient := new(http.Client)
+			httpClient.Timeout = 5 * time.Second
+
+			response, err := httpClient.Get("https://github.com/banzaicloud/koperator/releases/download/" + koperatorVersion + "/kafka-operator.crds.yaml")
```

Comment:
It would be better to use here this path: https://raw.githubusercontent.com/banzaicloud/koperator/$branchnameOrTag/config/base/crds/kafka.banzaicloud.io_kafkaclusters.yaml etc...
In this way we can run test for different branches